nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-api/src/main/java/org/apache/nifi/dbcp/DBCPService.java

```diff
@@ -31,4 +33,6 @@
 @CapabilityDescription("Provides Database Connection Pooling Service. Connections can be asked from pool and returned after usage.")
 public interface DBCPService extends ControllerService {
     public Connection getConnection()  throws ProcessException;
+
+    public DataSource getDataSource()  throws ProcessException;
```

Comment:
If we declare getDataSource() method with default implementation like below, we don't have to touch every existing implementation codes in test classes:

```
    default DataSource getDataSource()  throws ProcessException {
        throw new UnsupportedOperationException();
    }
```